src/operator/aggregate/average.rs

```diff
@@ -10,35 +13,74 @@ use std::{
     ops::{Add, AddAssign, Div, Neg},
 };
 
-/// Intermediate representation of an average as a `(sum, count)` pair.
-#[derive(Debug, Clone, Eq, Hash, PartialEq, Ord, PartialOrd, SizeOf)]
-pub struct Avg<T> {
+/// Representation of a partially computed average aggregate as a `(sum, count)`
+/// tuple.
+///
+/// This struct represents the result of the linear part of the average
+/// aggregate as a `(sum, count)` tuple (see [`Stream::average`]).  The actual
+/// average value can be obtained by dividing `sum` by `count`.  `Avg` forms a
+/// commutative monoid with point-wise plus operation `(sum1, count1) + (sum2,
+/// count2) = (sum1 + sum2, count1 + count2)`.
+#[derive(Debug, Default, Clone, Eq, Hash, PartialEq, Ord, PartialOrd, SizeOf)]
+pub struct Avg<T, R> {
     sum: T,
-    count: isize,
+    count: R,
 }
 
-impl<T> Avg<T> {
-    pub const fn new(sum: T, count: isize) -> Self {
+impl<T, R> Avg<T, R> {
+    /// Create a new `Avg` object with the given `sum` and `count`.
+    pub const fn new(sum: T, count: R) -> Self {
         Self { sum, count }
     }
+
+    /// Returns the `sum` component of the `(sum, count)` tuple.
+    pub fn sum(&self) -> T
+    where
+        T: Clone,
+    {
+        self.sum.clone()
+    }
+
+    /// Returns the `count` component of the `(sum, count)` tuple.
+    pub fn count(&self) -> R
+    where
+        R: Clone,
+    {
+        self.count.clone()
+    }
+
+    /// Returns `sum / count` or `None` if `count` is zero.
```

Comment:
If count is zero the sum better be zero too, and then 0 is also an acceptable result.
In SQL it is NULL anyway

Reply:
> If count is zero the sum better be zero too, and then 0 is also an acceptable result.

This isn't obvious to me. 0 seems like an arbitrary choice in this case.